src/examples/zkapps/joint-update.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * This is an example for two zkapps that are guaranteed to update their states _together_.
+ *
+ * So, A's state will updated if and only if B's state is updated, and vice versa.
+ *
+ * The difficulty here is that, while zkApps know and prove which other zkApps they call themselves,
+ * there's nothing in the protocol that lets them know which other zkApps are calling _them_.
+ *
+ * In other words, while one-way interactions are easy to implement, two-way interactions require additional tricks.
+ *
+ * This example is supposed to give you an idea for how to implement two-way interactions.
+ *
+ * The idea is that the user calls B, which calls A, so B knows it's jointly updating with A.
+ * In addition, B sets a flag "insideBUpdate" to true on its onchain state, which A checks to make sure it's being called from B.
+ * This flag is also reset by the method that A calls, so it's guaranteed to always be false when we're not inside a B update.
+ * That way, both A and B can be sure that they're updating together.
+ *
+ * To understand the flow of this example in detail, keep in mind that zkApp updates are applied top-to-bottom:
+ * 1. First, the account update created by `B.updateWithA()` is applied. It sets `insideBUpdate = true`.
+ * 2. Then, `A.updateWithB()` is applied.
+ * 3. Finally, `B.assertInsideUpdate()` is applied. It checks that `insideBUpdate = true` and sets it back to false.
+ */
+import {
+  Bool,
+  Field,
+  PrivateKey,
+  SmartContract,
+  State,
+  method,
+  state,
+} from 'o1js';
+
+const aPubKey = PrivateKey.randomKeypair().publicKey;
+const bPubKey = PrivateKey.randomKeypair().publicKey;
+
+class A extends SmartContract {
```

Comment:
[NIT] May be more clear to name these something like `Parent` and `Child`, or if that's too overloaded `Main` and `Dependent`, rather than A and B.  A and B makes me imagine siblings with which I can interact one or the other.  In fact, we can only interact with B.

Reply:
Good point!